internal/runtime/client/client.go

```diff
@@ -488,8 +488,13 @@ func httpCall(ctx context.Context, request, response runtime.Object, opts *httpC
 	})
 
 	resp, err := client.Do(httpRequest)
+	respStatus := "Unknown"
+
 	// Create http request metric.
-	runtimemetrics.RequestsTotal.Observe(httpRequest, resp, opts.hookGVH, err)
+	defer func() {
+		runtimemetrics.RequestsTotal.Observe(httpRequest, resp, opts.hookGVH, err, respStatus)
+	}()
```

Comment:
```suggestion
	defer runtimemetrics.RequestsTotal.Observe(httpRequest, resp, opts.hookGVH, err, respStatus)
```
nit


+1 to adding this to the `request_total`. Makes sense there.

Reply:
We should use a `defer func` there because we need access to `respStatus` variable, which can be updated later in the code flow.